app/src/main/java/com/conupods/IndoorMaps/IndoorBuildingOverlays.java

```diff
@@ -0,0 +1,232 @@
+import com.conupods.IndoorMaps.ConcreteBuildings.VLBuilding;
+import com.conupods.OutdoorMaps.Building;
+import com.conupods.R;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.BitmapDescriptor;
+import com.google.android.gms.maps.model.BitmapDescriptorFactory;
+import com.google.android.gms.maps.model.GroundOverlay;
+import com.google.android.gms.maps.model.GroundOverlayOptions;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.MapStyleOptions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class IndoorBuildingOverlays {
+
+    private GoogleMap mMap;
+
+    public enum Buildings{
+        HALL, MB, VL, CC
+    }
+
+    private static Building hInstance = HBuilding.getInstance();
+    private static Building mbInstance = MBBuilding.getInstance();
+    private static Building vlInstance = VLBuilding.getInstance();
+    private static Building ccInstance = CCBuilding.getInstance();
+
+    private List<BitmapDescriptor> mImages = new ArrayList<BitmapDescriptor>();
+    private GroundOverlay mHALLOverlay;
+    private GroundOverlay mMBOverlay;
+    private GroundOverlay mLOYCCOverlay;
+    private GroundOverlay mLOYVLOverlay;
+    private View mLevelButtons;
+    private View floorButtonsHall;
+    private View floorButtonsMB;
+    private View floorButtonsLOY_VL;
+
+    public IndoorBuildingOverlays(View LevelButtons, GoogleMap map) {
+        mMap = map;
+        mLevelButtons = LevelButtons;
+        mMap.setIndoorEnabled(false);
+
+        mImages.clear();
+
+        //index = 0 is first floor of Hall
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.h1));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.h2));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.h8));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.h9));
+
+        //index = 4 is first floor of MB
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.mb1));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.mbs2));
+
+        //index = 6 is first floor of LOY_CC
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.cc1));
+
+        //index = 7 is first floor of LOY_VL
+        mImages.add(BitmapDescriptorFactory.fromResource((R.drawable.loy_vl1)));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.loy_vl2));
+
+
+        floorButtonsMB =  mLevelButtons.findViewById(R.id.floorButtonsMB);
+        floorButtonsHall = mLevelButtons.findViewById(R.id.floorButtonsHall);
+        floorButtonsLOY_VL = mLevelButtons.findViewById(R.id.floorButtonsLOYVL);
+    }
+
+    public void hideLevelButton() {
+        floorButtonsMB.setVisibility(View.INVISIBLE);
+        floorButtonsHall.setVisibility(View.INVISIBLE);
+        floorButtonsLOY_VL.setVisibility(View.INVISIBLE);
+    }
+
+    public void showFloorButtons(Buildings building) {
```

Comment:
this should be `Buildings buildings` both plural no? Otherwise, it will be confused for a building entity - better yet name your enums BuildingCodes because that is exactly that they are!